scripts/un/sdg/util.py

```diff
@@ -0,0 +1,233 @@
+measuredProperty: dcs:value
+name: {name}
+populationType: dcs:{popType}
+statType: dcs:measuredValue{cprops}
+'''
+MMETHOD_TEMPLATE = '''
+Node: dcid:{dcid}
+typeOf: dcs:SDG_MeasurementMethodEnum
+name: "{dcid}"
+description: "{description}"
+'''
+UNIT_TEMPLATE = '''
+Node: dcid:{dcid}
+typeOf: dcs:UnitOfMeasure
+name: "{name}"
+description: "SDG Unit: {dcid}"
+'''
+
+# Select dimensinos will be modeled differently.
+SKIPPED_DIMENSIONS = {
+    'Cities', 'Freq', 'Nature', 'Observation Status', 'Report Ordinal',
+    'Reporting Type', 'UnitMultiplier', 'Units'
+}
+
+# Use existing properties when they exist.
+# TODO: Also map enums to existing nodes.
+MAPPED_DIMENSIONS = {
+    'Age': 'age',
+    'Cause of death': 'causeOfDeath',
+    'Disability status': 'disabilityStatus',
+    'Education level': 'educationalAttainment',
+    'Sex': 'gender',
+    'AGE': 'age',
+    'CAUSE_OF_DEATH': 'causeOfDeath',
+    'DISABILITY_STATUS': 'disabilityStatus',
+    'EDUCATION_LEVEL': 'educationalAttainment',
+    'SEX': 'gender'
+}
+
+BASE_DIMENSIONS = {
+    'SERIES_CODE', 'SERIES_DESCRIPTION', 'VARIABLE_CODE',
+    'VARIABLE_DESCRIPTION', 'GEOGRAPHY_CODE', 'GEOGRAPHY_NAME',
+    'GEOGRAPHY_TYPE', 'GEO_AREA_CODE', 'GEO_AREA_NAME', 'CITIES',
+    'SAMPLING_STATIONS', 'TIME_PERIOD', 'TIME_DETAIL', 'TIME_COVERAGE', 'FREQ',
+    'VALUE', 'VALUE_TYPE', 'UPPER_BOUND', 'LOWER_BOUND', 'UNITS',
+    'UNITMULTIPLIER', 'BASE_PERIOD', 'NATURE', 'SOURCE', 'GEO_INFO_URL',
+    'FOOT_NOTE', 'REPORTING_TYPE', 'OBSERVATION_STATUS', 'RELEASE_STATUS',
+    'RELEASE_NAME'
+}
+
+# Create map of M49 -> ISO-alpha3 for countries.
+with open(os.path.join(module_dir_, 'm49.csv')) as f:
+    PLACES = {}
+    reader = csv.DictReader(f, delimiter='\t')
+    for row in reader:
+        if not row['ISO-alpha3 code']:  # Only countries for now.
+            continue
+        PLACES[int(row['M49 code'])] = row['ISO-alpha3 code']
+
+# Create map of name -> dcid for supported cities.
+with open(os.path.join(module_dir_, 'cities.csv')) as f:
+    reader = csv.DictReader(f)
+    CITIES = {row['name']: row['dcid'] for row in reader}
+
+
+def format_description(s):
+    '''Formats input with curated style.
+
+    Args:
+        s: Input string.
+
+    Returns:
+        Curated string.
+    '''
+    # Remove <=2 levels of ().
+    formatted = re.sub('\((?:[^)(]|\([^)(]*\))*\)', '', s)
+    # Remove <=2 levels of [].
+    formatted = re.sub('\[(?:[^)(]|\[[^)(]*\])*\]', '', formatted)
+    # Remove attributes indicated with 'by'.
+    formatted = formatted.split(', by')[0]
+    # Remove references indicated by 'million USD'.
+    formatted = formatted.split(', million USD')[0]
+    # Remove extra spaces
+    formatted = formatted.replace(' , ', ', ').replace('  ', ' ').strip()
+    # Remove trailing commas
+    if formatted[-1] == ',':
+        formatted = formatted[:-1]
+    # Replace 100,000 with 100K
+    formatted = formatted.replace('100,000', '100K')
+    # Make ascii
```

Comment:
To confirm, are these the only ones, or should we use some encoding?

Reply:
[from offline discussion] these are the only non-ascii characters currently. we decided to keep the manual curation for now